x/ccv/provider/keeper/msg_server.go

```diff
@@ -87,36 +88,19 @@ func (k msgServer) AssignConsumerKey(goCtx context.Context, msg *types.MsgAssign
 	return &types.MsgAssignConsumerKeyResponse{}, nil
 }
 
-// ConsumerAddition defines an RPC handler method for MsgConsumerAddition
-func (k msgServer) ConsumerAddition(goCtx context.Context, msg *types.MsgConsumerAddition) (*types.MsgConsumerAdditionResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
-	}
-
-	ctx := sdk.UnwrapSDKContext(goCtx)
-	err := k.Keeper.HandleConsumerAdditionProposal(ctx, msg)
-	if err != nil {
-		return nil, errorsmod.Wrapf(err, "failed handling ConsumerAddition proposal")
-	}
-	return &types.MsgConsumerAdditionResponse{}, nil
-}
-
-// ConsumerRemoval defines an RPC handler method for MsgConsumerRemoval
-func (k msgServer) ConsumerRemoval(
+// RemoveConsumer defines an RPC handler method for MsgRemoveConsumer
+func (k msgServer) RemoveConsumer(
 	goCtx context.Context,
-	msg *types.MsgConsumerRemoval,
-) (*types.MsgConsumerRemovalResponse, error) {
+	msg *types.MsgRemoveConsumer) (*types.MsgRemoveConsumerResponse, error) {
 	if k.GetAuthority() != msg.Authority {
 		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	err := k.Keeper.HandleConsumerRemovalProposal(ctx, msg)
-	if err != nil {
-		return nil, errorsmod.Wrapf(err, "failed handling ConsumerAddition proposal")
-	}
 
-	return &types.MsgConsumerRemovalResponse{}, nil
+	k.Keeper.SetConsumerIdToStopTime(ctx, msg.ConsumerId, msg.StopTime)
```

Comment:
Does this mean I can set a stop time for a consumer chain that is not even launched yet?
Is there no way to remove the consumer chain record before it is launched?

Reply:
Yes, changed it to only be able to remove consumer if the chain is `Launched`.
> Is there no way to remove the consumer chain record before it is launched?
What would be the benefit of this?